src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -679,27 +682,27 @@ def filter_out_esm_packages(self, packages, package_versions):
         """
         Filter out packages from the list where the version matches the UA_ESM_REQUIRED string.
         """
-        non_esm_packages = []
-        non_esm_package_versions = []
-        esm_packages = []
-        esm_package_versions = []
-        esm_packages_found = False
+        selected_packages = []
+        selected_packages_versions = []
+        ua_esm_required_packages = []
+        ua_esm_required_package_versions = []
+        ua_esm_required_packages_found = False
 
         for pkg, version in zip(packages, package_versions):
             if version != Constants.UA_ESM_REQUIRED:
-                non_esm_packages.append(pkg)
-                non_esm_package_versions.append(version)
+                selected_packages.append(pkg)
+                selected_packages_versions.append(version)
                 continue
 
             # version is UA_ESM_REQUIRED.
-            esm_packages.append(pkg)
-            esm_package_versions.append(version)
+            ua_esm_required_packages.append(pkg)
+            ua_esm_required_package_versions.append(version)
 
-        esm_packages_found = len(esm_packages) > 0
-        if esm_packages_found:
-            self.status_handler.add_error_to_status("{0} patches requires Ubuntu Pro for Infrastructure with Extended Security Maintenance".format(len(esm_packages)), Constants.PatchOperationErrorCodes.UA_ESM_REQUIRED) # Set the error status with the esm_package details. Will be used in portal.
-            self.telemetry_writer.write_event("Filter esm packages [EsmPackagesCount={0}]".format(len(esm_packages)), Constants.TelemetryEventLevel.Informational)
+        ua_esm_required_packages_found = len(ua_esm_required_packages) > 0
+        if ua_esm_required_packages_found:
+            self.status_handler.add_error_to_status("{0} patches requires Ubuntu Pro for Infrastructure with Extended Security Maintenance".format(len(ua_esm_required_packages)), Constants.PatchOperationErrorCodes.UA_ESM_REQUIRED) # Set the error status with the esm_package details. Will be used in portal.
+            self.telemetry_writer.write_event("Filter esm packages [EsmPackagesCount={0}]".format(len(ua_esm_required_packages)), Constants.TelemetryEventLevel.Informational)
```

Comment:
Logger should take care of writing to telemetry. In the sense, line 706, will write this to log and to telemetry. See log_debug() in CompositeLogger.py. Line 704 is not needed

Reply:
Done